packages/core/src/utils/renderProps.tsx

```diff
@@ -35,7 +35,7 @@ export function renderProps<Type extends ElementType>(
 
   if (Type) {
     // Case 3: If render is not provided, render the Type component with the rest of the props
-    return <Type {...restProps} />;
+    return <Type key={key} {...restProps} />;
```

Comment:
Does this actually work? keys aren't treated as props so having renderProps support it won't work?

Reply:
was an issue but no longer used, so removed